plugins/Marketplace/resources/qml/OnboardBanner.qml

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2021 Ultimaker B.V.
+// Cura is released under the terms of the LGPLv3 or higher.
+
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.1
+
+import UM 1.6 as UM
+import Cura 1.6 as Cura
+
+// Onboarding banner.
+Rectangle
+{
+    property bool bannerVisible
+    property string bannerIcon
+    property string bannerBody
+    property var onRemoveBanner
```

Comment:
This should be a signal. That way you can just emit the "onBannerHidden" signal and connect any number functions to it from the outside. 

I would also put the visibility toggle in the banner itself (so the outside doesn't have to remember to actually hide it)

Reply:
> I would also put the visibility toggle in the banner itself (so the outside doesn't have to remember to actually hide it)

I tried that before. Unfortunately this did not result in the desired result. There is a single banner element for al three pages. If you hide a banner for one page, it should not be hidden from the remaining pages. Hiding the banner this way removes the banner element in its entirety; it also becomes hidden from the other pages.

Is there a way to "reset" the properties if the page changes?

Reply:
I also had a bit of difficulties using a signal as you suggested to remove the on boarding banners. This is due to the multiple pages where the banners are located; as there are 3 pages there are 3 places where we listen to the `onBannerHidden` signal. As all these the handlers for all these 3 pages are listening to the signal at the same time all 3 banners are removed by clicking on a single close button. There are several options for resolving this:
- create 3 different signals, a separate signal for each banner
- add a check within the signal listener to check of the signal is emitted from the right page.
I think these stated options add unnecessary complexity. Of course there is probably a very obvious solution, but as I'm new to Qt I'm just missing it. I gladly hear suggestions.

Reply:
Signals can have parameters, so you could also use that.

I tried something as well: Just saving the preference (location) string and moving the function to the OnboardBanner base class. I almost had that working but each time something else stopped working, so I decided that it was outside of the scope of this review & decided to accept the code as-is for now.